CMakeLists.txt

```diff
@@ -21,6 +21,12 @@ if (is_multiconfig STREQUAL "NOTFOUND")
   endif ()
 endif ()
 
+if (${CMAKE_CURRENT_SOURCE_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
```

Comment:
I think this is detecting in source builds vs. out of source builds. Is that correct? If so, I'd rename the variable from `is_root_project` to `is_insource_build`. If that's not correct can you explain what this is decting?

Reply:
this detects when we are being included in another project for the purpose of linking to `xrpl_core`. In that case, we only want to build/install the necessary core libraries and not rippled+deps. That use case is briefly described in  https://github.com/ripple/rippled/blob/09050a860be745c88a54fbcbce1e317c72f2f51e/Builds/linux/README.md#option-1-git-submodules--add_subdirectory.

Reply:
Thanks for clarifying this, I understand now. It's nothing to do with in source and out of source, it's if this is the "main" CMakeFile or it was included from another project. I was just confused.